flang/include/flang/Optimizer/Builder/Runtime/RTBuilder.h

```diff
@@ -400,6 +400,69 @@ constexpr TypeBuilderFunc getModel<bool &>() {
     return fir::ReferenceType::get(f(context));
   };
 }
+template <>
+constexpr TypeBuilderFunc getModel<unsigned short>() {
+  return getModel<short>();
```

Comment:
The existing code does not distinguish `unsigned int` from `signed int`, so I followed that precedent for other sizes of `unsigned` types.  Should the patch be changed to mark the signature for `unsigned int` as being unsigned, along with the other sizes, or just the other sizes?

Reply:
> The existing code does not distinguish unsigned int from signed int
Oh, I see.

That is likely just because only the short case is likely problematic. The [code dealing with target specific aspects](https://github.com/llvm/llvm-project/blob/ed6ddffb583beb450c4b0e1747ccd14f7e063105/flang/lib/Optimizer/CodeGen/Target.cpp#L113 ) that adds signext/zeroext is only triggered for integer types strictly smaller that the `int` type.

I think I would favor things being consistent and having the unsigned aspect always set here and leave it up to the target aspects to do whatever is needed when generating the LLVM IR (that does not have an unsigned integer type, just specific operations and integer extension attributes).

Reply:
I did try marking all of these explicitly unsigned where appropriate and it broke the world in ways I didn't understand.  We can look at this further later when/if I merge this feature and try to clean it up.

Reply:
Looks like this is "only" a lit test fallout (the generated FIR/LLVM IR is now different for 65 lit tests that test code that involve calling some runtime function with an unsigned argument). I agree it is best to fix that in a different PR (the behavior is already there).